test/core/test/repro.test.ts

```diff
@@ -0,0 +1,235 @@
+  },
+})
+
+myTest.concurrent.each2([['hello'], ['hi']])(
+  'myTest.each2 %s',
+  (...args) =>
+    async ({ expect, myFixture }) => {
+      await sleep(200)
+      expect({ args, myFixture }).toMatchSnapshot()
+    },
+)
+
+myTest.concurrent.each3([['hello'], ['hi']])(
+  'myTest.each3 %s',
+  async ({ expect, myFixture, args }) => {
+    await sleep(200)
+    expect({ args, myFixture }).toMatchSnapshot()
+  },
+)
+
+myTest.concurrent.each4([['hello'], ['hi']])(
+  'myTest.each4 %s',
+  async (arg, { expect, myFixture }) => {
+    await sleep(200)
+    expect({ arg, myFixture }).toMatchSnapshot()
+  },
+)
+
+// type test
+;() => myTest.concurrent.each5([['hello'], ['hi']])(
+  'myTest.each5-context-false %s',
+  // @ts-expect-error no context
+  async (arg, { expect }) => {
+    await sleep(200)
+    expect(arg).toBeTypeOf('string')
+  },
+)
+
+myTest.concurrent.each5([['hello'], ['hi']])(
+  'myTest.each5-context-false %s',
+  async (arg) => {
+    await sleep(200)
+    expect(arg).toBeTypeOf('string')
+  },
+)
+
+myTest.concurrent.each5([['hello'], ['hi']], { context: true })(
+  'myTest.each5-context-true %s',
+  async (arg, { expect, myFixture }) => {
+    await sleep(200)
+    expect({ arg, myFixture }).toMatchSnapshot()
+  },
+)
+
+myTest.concurrent.each5([['hello', 2], ['hi', 3]])(
+  'myTest.each5-context-false-multiple %s %s',
+  async (argString, argNumber) => {
+    await sleep(200)
+    expect(argString).toBeTypeOf('string')
+    expect(argNumber).toBeTypeOf('number')
+    argString satisfies string
+    argNumber satisfies number
+  },
+)
+
+myTest.concurrent.each5([['hello', 2], ['hi', 3]])(
+  'myTest.each5-context-false-multiple-spread %s %s',
+  async (...args) => {
+    await sleep(200)
+    expect(args[0]).toBeTypeOf('string')
+    expect(args[1]).toBeTypeOf('number')
+    args[0] satisfies string
+    args[1] satisfies number
+  },
+)
+
+myTest.concurrent.each5([['hello'], ['hi']], { context: true })(
+  'myTest.each5-context-true %s',
+  async (arg, { expect, myFixture }) => {
+    await sleep(200)
+    expect({ arg, myFixture }).toMatchSnapshot()
+  },
+)
+
+myTest.concurrent.each5([['hello', 2], ['hi', 3]], { context: true })(
+  'myTest.each5-context-true-multiple %s %s',
+  async (argString, argNumber, { expect, myFixture }) => {
+    await sleep(200)
+    expect({ argString, argNumber, myFixture }).toMatchSnapshot()
+    expect(argString).toBeTypeOf('string')
+    expect(argNumber).toBeTypeOf('number')
+    argString satisfies string
+    argNumber satisfies number
+  },
+)
+
+// cannot support spread with context
+;() => myTest.concurrent.each5([['hello', 2], ['hi', 3]], { context: true })(
+  'myTest.each5-context-true-multiple-spread %s %s',
+  async (...args) => {
+    await sleep(200)
+    expect(args[0]).toBeTypeOf('string')
+    expect(args[1]).toBeTypeOf('number')
+    args[0] satisfies string
+    args[1] satisfies number
+  },
+)
+
+myTest.concurrent.each6([['hello'], ['hi']], { context: true })(
```

Comment:
I posted a writeup in https://github.com/vitest-dev/vitest/issues/4642#issuecomment-1898866866, but I suspect that `each6` here is the most compatible option. Note that you can also write `{ expect }, ...args` with this form, but you can't with `each5`.

Reply:
There is also another syntax for `each` using the template literal:

```ts
test.each`
  name | description 
  ${1} | ${"one"}
`("test", () => {})
```

So I am not sure we can have a second argument in each.

Reply:
Oh, right, that yucky thing :(